src/batchpage.js

```diff
@@ -0,0 +1,125 @@
+    this.state = {
+      terms: args.terms ? ("" + args.terms).replace(/\+/g, ' ') : "",
+      args: args,
+      collections: null,
+      selectedCollections: null,
+      limits: null,
+    }
+  }
+
+  componentDidMount() {
+    this.getCollectionsAndLimits()
+  }
+
+  getCollectionsAndLimits() {
+    $.get('/collections', function (resp) {
+      var collections = resp
+      var selectedCollections = null
+      var args = this.state.args
+      if (args.collections) {
+        var sel = '' + args.collections
+        selectedCollections = sel ? sel.split('+') : []
+      }
+      else {
+        selectedCollections = resp.map((c) => c.name)
+      }
+
+      this.setState({
+        collections,
+        selectedCollections,
+      })
+    }.bind(this))
+
+    $.get('/limits', function(resp) {
+      this.setState({limits: resp})
+    }.bind(this))
+  }
+
+  buildQuery(termsString, selectedCollections, batchSize) {
+    if(! termsString.trim()) return null
+    if(! selectedCollections.length) return null
+
+    return {
+      terms: termsString.trim().split('\r\n'),
+      collections: selectedCollections,
+      batchSize: batchSize,
+    }
+  }
+
+  render() {
+    if(! (this.state.collections && this.state.limits)) {
+      return <p>loading ...</p>
+    }
+
+    let refreshForm = () => {
+      if(this.refs.terms.value)
+        this.refs.form.submit()
+    }
+
+    let onChangeCollections = (selected) => {
+      this.setState({selectedCollections: selected})
+      setTimeout(refreshForm, 0)
+    }
+
+    let collectionsValue = this.state.selectedCollections.join(' ')
+    let {terms, collections, selectedCollections, limits} = this.state
+    let batchSize = limits.batch
+    let query = this.buildQuery(terms, selectedCollections, batchSize)
+
+    return (
+      <form id="batch-form" ref="form">
+        <input type="hidden" name="collections" value={collectionsValue} />
+        <div className="row">
+          <div className="col-sm-2">
+            <h1>Hoover</h1>
+            <CollectionsBox
+              collections={collections}
+              selected={selectedCollections}
+              onChange={onChangeCollections} />
+          </div>
+          <div className="col-sm-9">
+            <div id="batch-input-box" className="form-group">
+              <textarea
+                id="batch-input-terms"
+                ref="terms"
+                name="terms"
+                className="form-control"
+                rows="8"
+                placeholder="search terms, one per line"
+                defaultValue={terms}
+                ></textarea>
+            </div>
+            <div id="search-infotext" className="form-text text-muted">
+              <p id="search-back">
+                <a href="./">Back to single search</a>
+              </p>
+            </div>
+            <div className="form-inline">
+              <button type="submit" className="btn btn-primary">
```

Comment:
`limits.requests != null` and the number of lines inserted is greater than `limits.batch * limits.requests.count`, then disable the button (and maybe make the above text red)

Reply:
Or if you're bored, check the `/limits` endpoint before actually sending the query, so the request fails here instead of having the list halfway rejected.